### DISCUSSION TITLE
Disable validations Per environments (i.e. do not validate on Production)

### DISCUSSION THREAD

**Discussion by galarbel** [colinhacks/zod#1924] [Q&A] [answered]
Hi All,

this looks like a really great project, but I was looking for a way to set it up so that real-time validations will only be done on Non-Production environments.

the idea is to make sure that production environment will run as fast as possible.
similar to what redux have with their "production mode".

is there any way to achieve this?
are there any plans to add something like that?

Thanks,
Gal

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
// for node use: process.env.ENVIRONMENT
const schema = Deno.env.get( 'ENVIRONMENT' ) === 'production'
    ? z.any()
    : z.string()
```

**Reply by dmnd** [reply]
This is the behaviour I'm looking for in a very narrow sense. But it's not practical in a real app:

- it destroys type safety in non production environments
- it requires a code change to every schema definition call site
- to write a proxy to wrap around zod and take care of the two above issues, you need a list of every zod type to override

it'd be much more practical if there was a built-in way to globally configure zod like OP suggested, e.g.

```ts
if (process.env.NODE_ENV === "production") {
  z.unsoundlyIgnoreSchemas();
}
```

**Comment by galarbel** [comment]
something like this just might work :)
it'd be nice to see something like this built in though.

`z.configure({ production: true });`

this could potentially help with performance issues as was mentioned here:
https://github.com/colinhacks/zod/issues/205

thanks for the quick response.
